Tasks/DotNetCoreInstallerV1/versioninstaller.ts

```diff
@@ -0,0 +1,167 @@
+import * as path from 'path';
+import * as fs from "fs";
+import * as url from "url";
+
+import * as tl from 'vsts-task-lib/task';
+import * as toolLib from 'vsts-task-tool-lib/tool';
+
+import * as utils from "./versionutilities";
+import { VersionInfo } from './versionfetcher';
+
+export class VersionInstaller {
+    constructor(packageType: string, installationPath: string) {
+        try {
+            tl.exist(installationPath) || tl.mkdirP(installationPath);
+        }
+        catch (ex) {
+            throw tl.loc("UnableToAccessPath", installationPath, JSON.stringify(ex));
+        }
+
+        this.packageType = packageType;
+        this.installationPath = installationPath;
+    }
+
+    public async downloadAndInstall(versionInfo: VersionInfo, downloadUrl: string): Promise<void> {
+        if (!versionInfo || !versionInfo.version || !url.parse(downloadUrl)) {
+            throw tl.loc("VersionCanNotBeDownloadedFromUrl", versionInfo, downloadUrl);
+        }
+
+        let downloadPath = "";
+        let version = versionInfo.version;
+
+        try {
+            downloadPath = await toolLib.downloadTool(downloadUrl);
+        } catch (error) {
+            throw tl.loc("CouldNotDownload", downloadUrl, JSON.stringify(error));
+        }
+
+        try {
+            //todo: if installation path is outside agents directory, acquire Lock for installation and start timer of 10-20 minutes, after which lock shall be auto released.
```

Comment:
why is this required?

Reply:
This can be required when installing in a location outside of agents directory like C:\pf 
There might be multiple agents trying to install at the same location, this can help avoiding problems with that